### ISSUE TITLE
Gatsbyjs with lando - Expose port 8000

### ISSUE THREAD

**Issue by alejodaraio** [lando/lando#1105]
Hey guys, i was shooting to Up a nodejs server with Gatsbyjs, but im getting some problems to see my website in the browsers.

This is my current lando:

```yml
name: mysite

proxy:
  appserver:
    - mysite.lndo.site

services:

  appserver:
    type: node:8.9
    overrides:
      services:
        ports:
          - "8000:8000"
    globals:
      gatsby-cli: "latest"

tooling:
  npm:
    service: appserver
  node:
    service: appserver
  gatsby:
    service: appserver
```
After startup my lando i run the gatsby command to startup my App as developer.

```
lando npm run develop
```

> As extra info Gatsbyjs startup the app with localhost:8000

But! when i go to **mysite.lndo.site** i get bad gateway response. I was looking some issues about how to expose ports, i tried disable the proxy but i can't see what is worng.

I tried **mysite.lndo.site:8000** but this doesn't exists.

**Comment by dustinleblanc** [comment]
@alejodaraio try this:

```yml
proxy:
  appserver:
    - mysite.lndo.site:8000
```

That should theoretically work so long as gatsby is propperly running on the port. You'll need to provide a command to get it running

**Comment by AaronFeledy** [comment]
This is how I got mine working:

```
name: mysite

proxy:
  nodejs:
    - mysite.lndo.site:8000

services:
  nodejs:
    type: node:8.9
    overrides:
      services:
        ports: [8000:8000]
    globals:
      gatsby-cli: "next"

tooling:
  npm:
    service: nodejs
  node:
    service: nodejs
  gatsby:
    service: nodejs
```

With the above .lando.yml, `lando gastby develop` still would not host outside of the container. It seems that you need to tell gatsby to host outside of the container by specifying the host as 0.0.0.0 when running the develop command. You can do this by running `lando gatsby develop --host 0.0.0.0`. Starting it in that way will make your site available on mysite.lndo.site:8000 on your host machine.

**Comment by tylerssn** [comment]
@alejodaraio & @AaronFeledy here is what I finally ended up with after a quick chat with @dustinleblanc & @pirog.

```yaml
name: gatsby

proxy:
  appserver:
    - gatsby.lndo.site

services:
  # Prefer appserver as a name since Lando defaults to it - making `lando ssh` functional as opposed to a required service name (`lando ssh nodejs`
  appserver:
    type: node:9
    # One-time build step
    install_dependencies_as_me:
      - yarn global add gatsby-cli
      # Initial build fail-safe for the Sharp NodeJS package.
      #  Sharp can be a pain. If installed on host, it may break in Lando.
      - rm -rf node_modules && yarn
    run_as_root:
      # Considering port 443 & 80, grant non-root access to lower port numbers
      - setcap 'cap_net_bind_service=+ep' `which node`
    command: gatsby develop --host 0.0.0.0 --port 80

####
# Irrelevant, but bonus tooling

tooling:
  yarn:
    service: appserver
  npm:
    cmd: Nope, try `lando yarn` instead.
    service: appserver
  node:
    service: appserver
  gatsby:
    service: appserver
  gatsby:start:
    cmd: gatsby develop --host 0.0.0.0 --port 80
    service: appserver
```

**Comment by pirog** [comment]
Going to close this out but wanted to mention that in RC2 you can now set the default http port for a node service, this would alleviate the need for a custom proxy port. 

Here are the docs (and examples) for both the node port and proxy
**node**
https://docs.devwithlando.io/tutorials/node.html#setting-a-port
https://github.com/lando/lando/tree/master/examples/node

**proxy**
https://docs.devwithlando.io/config/proxy.html#routing-to-a-different-port
https://github.com/lando/lando/tree/master/examples/proxy

**Comment by tylerssn** [comment]
For anyone who arrives here from Google, the gist of what it takes to set up hot reloading within Lando is this:

1. The hot reload server needs to be bound to:
   - Host: 0.0.0.0
   - Port: 80
     > **Note** this is the default port that Lando test connectivity against when establishing the proxy. Otherwise, you need to add `port:` to your Node service in `.lando.yml`
2. The hot reload server needs to honor connections request via any hostname or otherwise configured to honor connections from `{node-proxy}.lndo.site`.

Here's an [example HMR for Laravel w/ Lando](https://github.com/lando/lando/issues/1105#issuecomment-458339719).

**Comment by benjifisher** [comment]
Hot reloading in Gatsby Just Works (:tm:) for me.

Maybe something in Lando has been fixed since the earlier comments on this issue. Maybe something has changed in Gatsby since then, too: I did not have to modify `gatsby-node.js` as @tylerssn suggested. I *do* have to start the Gatsby development server with

```
lando gatsby develop --host 0.0.0.0
```

For more information, see my little repo [Lando Gatsby Drupal](https://gitlab.com/benjifisher/lando-gatsby-drupal), which has my complete `.lando.yml`, and [Issue 30](https://gitlab.com/benjifisher/lando-gatsby-drupal/-/issues/30) on that repo, which directed me here.

There is a little screencast attached to that issue. I would upload it here, too, but GitHub does not seem to support `.webm` files. :-(
